doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,148 @@
+
+.. code-block::
+
+	ConnectedNodes(src, dst) :- Links(src, dst, true).
+	ConnectedNodes(src, dst) :- ConnectedNodes(src, intermediate_node),
+				Links(intermediate_node, dst, true), (src != dst).
+
+As we have changed the DDlog program we need to recompile it. However, it would be annoying to run commands that populate :code:`Links` relation every time we change our DDlog program. To resolve this nuisance we can embed some records into the DDlog program itself to serve as ground truth or initial state. Let's do all of that.
+
+.. code-block::
+
+	$ cd ../ && cat t1_reachibility_monitor.dl
+	input relation Links(src: string, dst: string, link_status: bool)
+	output relation ConnectedNodes(str: string, dst: string)
+
+	ConnectedNodes(src, dst) :- Links(src, dst, true).
+	ConnectedNodes(src, dst) :- ConnectedNodes(src, intermediate_node),
+				Links(intermediate_node, dst, true), (src != dst).
+	
+	Links("Menlo Park", "Santa Barbara", true).
+	Links("Menlo Park", "Salt Lake City", true).
+	Links("Santa Barbara", "Los Angeles", true).
+	Links("Los Angeles", "Santa Barbara", true).
+	Links("Los Angeles", "Salt Lake City", true).
+	Links("Salt Lake City", "Los Angeles", true).
+
+	$ ddlog -i t1_reachibility_monitor.dl
+	$ cd t1_reachibility_monitor_ddlog/ && cargo build --release
+	$ ./target/release/t1_reachibility_monitor_cli
+	<clip>
+	>> dump ConnectedNodes;
+	ConnectedNodes{.src = "Los Angeles", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Los Angeles", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Salt Lake City"}
+	ConnectedNodes{.src = "Menlo Park", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Salt Lake City", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Salt Lake City", .dst = "Santa Barbara"}
+	ConnectedNodes{.src = "Santa Barbara", .dst = "Los Angeles"}
+	ConnectedNodes{.src = "Santa Barbara", .dst = "Salt Lake City"}
+
+Perfect! Now it looks exactly as what we expected! This is a good time to start doing what this tutorial promised and start writing some C code! We are going to start with something simple yet important. Our initial C program will connect with DDlog program, insert one additional record to :code:`Links` relation, and print the content of :code:`ConnectedNodes` relation. Let's create :code:`reachibility_monitor.c` next to our DDlog program's code. The code is available in the provided :code:`reachibility_monitor_p1.c` file. The further discussion will refer to specific lines in that code.

```

Comment:
"as we expected?"

Reply:
Before doing that do you want to show the incremental nature of DDlog by changing a link status in the CLI?